src/gwt/src/org/rstudio/studio/client/workbench/views/environment/model/MemoryUsage.java

```diff
@@ -49,6 +74,47 @@ public final int getPercentUsed()
     */
    public final int getProcessPercentUsed()
    {
+      if (useProcessLimit())
+         return getPercentUsed();
       return (int)Math.round(((getProcess().getKb() * 1.0) / (getTotal().getKb() * 1.0)) * 100);
    }
+
+   /**
+    * If the session process has a specific limit, that's different than
+    * the total, use it for the percentage.
+    */
+   public final boolean useProcessLimit()
+   {
+      long limit = getLimit().getKb();
+      return limit != 0; // && limit != getTotal().getKb();
+   }
+
+   private final String limitMessage()
+   {
+      return useProcessLimit() ? constants_.limit() + ": " + StringUtil.prettyFormatNumber(getLimit().getKb()/1024) + " " + constants_.megabytes() : constants_.unlimited();
+   }
+
+   private final int getPercentFree()
+   {
+      return ((int) (100 * (getTotal().getKb() - getUsed().getKb()) / getTotal().getKb()));
+   }
+
+   public final String multiLineStatusMessage()
+   {
+      return constants_.sessionMemoryUsed() + ": " + StringUtil.prettyFormatNumber(getProcess().getKb()/1024) +
+	     " " + constants_.megabytes() + ", " + limitMessage() + "\n" +
+             constants_.freeSystemMemory() + ": " + StringUtil.prettyFormatNumber((getTotal().getKb() - getUsed().getKb())/1024) +
+	     " " + constants_.megabytes() + " (" + getPercentFree() + "%)";
+   }
+
+   public final String statusMessage()
+   {
+      return constants_.sessionMemoryUsed() + ": " + StringUtil.prettyFormatNumber(getProcess().getKb()/1024) +
```

Comment:
Composing UI strings this way is not i18n friendly as it assumes other languages have the same sentence ordering as English. Ideally, use numbered placeholders and perform the substitions in a single operation.

Also, having the colons `":"` be hardcoded is also a localization problem; all parts of the string should be coming from the string table.

An example of placeholders is this string:

https://github.com/rstudio/rstudio/blob/d2617df64ad85ea6719683bd472552f7243cac54/src/gwt/src/org/rstudio/studio/client/workbench/ui/UIConstants_en.properties#L4

Used here:

https://github.com/rstudio/rstudio/blob/d2617df64ad85ea6719683bd472552f7243cac54/src/gwt/src/org/rstudio/studio/client/workbench/ui/OptionsLoader.java#L111

Reply:
Thanks, these should be fixed now.